architecture/2024-11-authz-feature-spec.md

```diff
@@ -0,0 +1,390 @@
+
+> [!CAUTION]
+>
+> Ideally, the Radius administrator could specify a scope for the resource types they each role is able to administer within each resource group. Given the recipes are a property of environments today, it's unclear if this is feasible within the current design. 
+
+**Exceptions**
+
+The operation fails and informs the user interactively if:
+
+* The current user is not a member of the `radius-admin` or `group-admin` roles or the `cluster-admin` RoleBinding in the Kubernetes cluster
+* The role assignment already exists
+
+#### User story 7 – Deleting a role definition and assignment
+
+As a platform engineer at a large enterprise, I need to delete a role definition and assignment I previously created.
+
+**User Experience**
+
+```bash
+# Delete the previously defined role
+rad role definition delete my-role-definition
+# Error is shown because existing role assignments for this definition exist
+ERROR: The following role assignments exist for the my-role-definition role definition
+ROLE                ASSIGNEE            SCOPE
+my-role-definition  dba@my-company.net  /planes/radius/MyCompany/resourceGroup/*
+# Delete the role assignment
+rad role assignment delete \
+  --assignee dba@my-company.net \
+  --role recipe-admin \
+  --scope /planes/radius/MyCompany/resourceGroup/*
+# Delete the role definition
+rad role definition delete my-role-definition
+```
+
+**Result**
+
+1. The role assignment for the `my-role-definition` role is removed from the `dba@my-company.net` group
+2. The `my-role-definition` role definition is deleted
+
+**Exceptions**
+
+The operation fails and informs the user interactively if:
+
+* The current user is not a member of the `radius-admin` or `group-admin` roles or the `cluster-admin` RoleBinding in the Kubernetes cluster
+* Role assignments for the role definition exist as shown in the example
+
+## Feature Summary
+
+| Priority | Size | Feature                                                      |
+| -------- | ---- | ------------------------------------------------------------ |
+| p0       | S    | Out-of-the-box role definitions                              |
+| p0       | L    | UCP validates membership in either a role definition or the Kubernetes cluster-admin RoleBinding on each API call |
+| p0       | S    | Create or delete a role assignment for an individual user from the configured IdP system for a resource group (`rad role assignment create --assignee user@mycompany.net`) |
+| p1       | S    | Create or delete a role assignment for a group from the configured IdP system for a resource group (`rad role assignment create --assignee my-group@mycompany.net`) |
+| p1       | M    | Ability to create, update, and delete a role definition      |
+| p2       | M    | Replace the `default` resource group with `default-application` and `default-environment` and modify `rad environment`, `rad deploy`, and `rad run` to use these new groups |
```

Comment:
I really want to dig into this more.

Reply:
@zachcasper can you update this since we decided not to have a default application resource group?

Reply:
Updated